Mathlib/Order/Booleanisation.lean

```diff
@@ -0,0 +1,263 @@
+/-
+Copyright (c) 2023 Yaël Dillies. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yaël Dillies
+-/
+import Mathlib.Order.Hom.Lattice
+
+/-!
+# Adding complements to a generalized Boolean algebra
+
+This file embeds any generalized Boolean algebra into a Boolean algebra.
+
+This concretely proves that any equation holding true in the theory of Boolean algebras that does
+not reference `ᶜ` also holds true in the theory of generalized Boolean algebras. Put another way,
+one does not need the existence of complements to prove something which does not talk about
+complements.
+
+## Main declarations
+
+* `Booleanisation`: Boolean algebra containing a given generalised Boolean algebra as a sublattice.
+* `Booleanisation.inlLatticeHom`: Boolean algebra containing a given generalised Boolean algebra as
+  a sublattice.
```

Comment:
```suggestion
  a sublattice.
  
## Future work

Show that `inlLatticeHom` is the free functor from `GenBoolAlg` to `BoolAlg`.
```

Reply:
Added! But note that `Booleanisation` is the (object part of) the functor, not `inlLatticeHom`.